internal/extension/manager.go

```diff
@@ -140,57 +140,55 @@ func newExtensionService(dag *nilGuardedPointer[StateAwareDAG]) extpb.ExtensionS
 	return &extensionService{dag: dag}
 }
 
-func (d *StateAwareDAG) FindGatewaysFor(targetRefs []*extpb.TargetRef) ([]*extpb.Gateway, error) {
-	chain := d.topology.Objects().Items(func(o machinery.Object) bool {
-		return len(lo.Filter(targetRefs, func(t *extpb.TargetRef, _ int) bool {
-			return t.Name == o.GetName() && t.Kind == o.GroupVersionKind().Kind && t.Group == o.GroupVersionKind().Group
-		})) > 0
-	})
-
-	gateways := make([]*extpb.Gateway, 0)
-	chainSize := len(chain)
-
-	for i := 0; i < chainSize; i++ {
-		object := chain[i]
-		parents := d.topology.Objects().Parents(object)
-		chain = append(chain, parents...)
-		chainSize = len(chain)
-		if gw, ok := object.(*machinery.Gateway); ok && gw != nil {
-			gateways = append(gateways, toGw(*gw))
+func (s *extensionService) Subscribe(_ *emptypb.Empty, stream grpc.ServerStreamingServer[extpb.Event]) error {
+	for {
+		time.Sleep(time.Second * 5)
+		if err := stream.Send(&extpb.Event{}); err != nil {
+			return err
 		}
 	}
-
-	return gateways, nil
 }
 
-func toGw(gw machinery.Gateway) *extpb.Gateway {
-	return &extpb.Gateway{
-		Metadata: &extpb.Metadata{
-			Name:      gw.Gateway.Name,
-			Namespace: gw.Gateway.Namespace,
-		},
-		GatewayClassName: string(gw.Gateway.Spec.GatewayClassName),
-		Listeners:        toListeners(gw.Gateway.Spec.Listeners),
+func (s *extensionService) Resolve(_ context.Context, request *extpb.ResolveRequest) (*extpb.ResolveResponse, error) {
+	dag := BlockingDAG.getWait()
```

Comment:
This guarantees a consistent snapshot of the dag through our the whole resolve. But also means I can't keep `prg` around _as things currently are_. So I might just swap the `dag` out with a newer version on subsequent evaluations.